src/Bonus.java

```diff
@@ -38,29 +46,39 @@ public static void main(String[] args) {
         - Schrijf vervolgens een for-loop om de items in de hashset een voor een aan de String variabele toe te voegen.
         - Return de (gevulde) String variabele
          */
-    }
 
+        StringBuilder stringBuilder = new StringBuilder();
+        for (Integer number : hashSet) {
+            stringBuilder.append(number);
+        }
+        return stringBuilder.toString();
+    }
 
 
-    public static void/*moet dit "void" zijn of wat anders?*/ feedback(String stringnumber) {
+    public static void feedback(String stringnumber) {
         Scanner scanner = new Scanner(System.in);
         StringBuilder feedback = new StringBuilder();
+        Boolean isCorrect = false;
         System.out.println("+ = juiste nummer op de juiste plek, O = juiste nummer verkeerde plek, X = verkeerde nummer");
         System.out.println("Doe een gok, Let op vul 4 getallen in.");
-        String guess = scanner.nextLine();
-        if (Objects.equals(guess, stringnumber)) {
-            System.out.println("gefeliciteerd je hebt het goed");
-        } else {
-            for (int i = 0; i < 4; i++) {
-                if (guess.substring(i, i + 1).equals(stringnumber.substring(i, i + 1))) {
-                    feedback.append("+");
-                } else if (stringnumber.contains(guess.substring(i, i + 1))) {
-                    feedback.append("0");
-                } else {
-                    feedback.append("X");
+        while (!isCorrect) {
```

Comment:
isCorrect is een logische benaming, maar vind 'm persoonlijk lastig lezen. Je moet een extra denkstap maken dat deze false geinitialiseerd wordt en de negatief ervan dus true is. Mijn voorkeur zou zijn om een benaming als 'boolean continue = true' te doen die false wordt als het spel klaar is.